camel/models/schema_model.py

```diff
@@ -0,0 +1,233 @@
+)
+
+from openai import Stream
+from pydantic import BaseModel, ValidationError
+
+from camel.messages import OpenAIMessage
+from camel.models import BaseModelBackend
+from camel.types import (
+    ChatCompletion,
+    ChatCompletionChunk,
+    ChatCompletionMessage,
+    Choice,
+    ModelPlatformType,
+    ModelType,
+)
+from camel.utils import (
+    BaseTokenCounter,
+    OpenAITokenCounter,
+)
+
+T = TypeVar('T', bound=BaseModel)
+
+
+class SchemaModel(BaseModelBackend):
+    r"""Shema model in a unified BaseModelBackend interface, which aims to
+    generate the formatted response."""
+
+    def __init__(
+        self,
+        model_platform: ModelPlatformType,
+        model_type: str,
+        model_config_dict: Dict[str, Any],
+        url: Optional[str] = None,
+    ) -> None:
+        r"""Constructor for OpenAI backend.
+
+        Args:
+            model_platform (ModelPlatformType): Platform from which the model
+                originates, including transformers, llama_cpp, and vllm.
+            model_type (str): Model for which a backend is created, for
+                example, "mistralai/Mistral-7B-v0.3".
+            model_config_dict (Dict[str, Any]): A dictionary that will
+                be fed into openai.ChatCompletion.create().
+            url (Optional[str]): The url to the OpenAI service.
+        """
+        from outlines import models  # type: ignore[import]
+
+        self.model_platform = model_platform
+        self.model_name = model_type
+        self.model_config_dict = model_config_dict
+        self._client = Union[models.Transformers, models.LlamaCpp, models.VLLM]
+        self._url = url
+
+        # Since Outlines suports multiple model types, it is necessary to
+        # read the documentation to learn about the model kwargs:
+        # https://outlines-dev.github.io/outlines/reference/models/transformers
+        if self.model_platform == ModelPlatformType.OUTLINES_TRANSFORMERS:
+            model_kwargs = self.model_config_dict.get("model_kwargs", {})
+            device = self.model_config_dict.get("device", None)
+            tokenizer_kwargs = self.model_config_dict.get(
+                "tokenizer_kwargs", {}
+            )
+
+            self._client = models.transformers(
+                model_name=self.model_name,
+                device=device,
+                model_kwargs=model_kwargs,
+                tokenizer_kwargs=tokenizer_kwargs,
+            )
+        elif self.model_platform == ModelPlatformType.OUTLINES_LLAMACPP:
+            repo_id = self.model_config_dict.get(
+                "repo_id", "TheBloke/phi-2-GGUF"
+            )
+            filename = self.model_config_dict.get(
+                "filename", "phi-2.Q4_K_M.gguf"
+            )
+            download_dir = self.model_config_dict.get("download_dir", None)
+            model_kwargs = self.model_config_dict.get("model_kwargs", {})
+
+            from llama_cpp import llama_tokenizer  # type: ignore[import]
```

Comment:
1. `llama-cpp-python` is not added to `pyproject.toml`?
2. The same issue on `# type: ignore[import]` as that for `outlines` package.

Reply:
I will remove outlines from pyproject.toml